third_party/move/move-prover/bytecode-pipeline/src/number_operation_analysis.rs

```diff
@@ -461,15 +461,41 @@ impl<'a> NumberOperationAnalysis<'a> {
                                 for (arg, arg_oper) in args.iter().zip(arg_oper.iter()) {
                                     if merged != *arg_oper {
                                         // propagate to arg if necessary
-                                        global_state.update_node_oper(
-                                            arg.node_id(),
-                                            merged,
-                                            allow_merge,
+                                        let mut update_flag = matches!(
+                                            arg.clone().into(),
+                                            ExpData::Temporary(..)
+                                                | ExpData::LocalVar(..)
+                                                | ExpData::Value(..)
+                                                | ExpData::Call(_, ASTOperation::Cast, _)
                                         );
-                                        update_temporary(arg, &merged, global_state, state);
+                                        if let ExpData::Call(
```

Comment:
It would probably simplify the code if you used a match expression and combined this with the previous statement, e.g.:
```
    let update_flag = match arg.clone().into() {
        ExpData::Temporary(..) | ... => true,
        ExpData::Call(_, ASTOperation::SpecFunction(mid, sid, _), _) => {
            let module_env = ..
            ...
            !spec_f.is_move_fun && self.func_target...
        },
        _ => false,
     }
```

Reply:
done